exampleAppJapan/src/main/kotlin/it/trade/android/japanapp/ui/orderinput/OrderInputViewModel.kt

```diff
@@ -0,0 +1,158 @@
+package it.trade.android.japanapp.ui.orderinput
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+
+class OrderInputViewModel : ViewModel() {
+    // TODO: Implement the ViewModel
+
+    private lateinit var orderForm: MutableLiveData<OrderForm>
+
+    fun getOrderModel(): LiveData<OrderForm> {
+        if (!this::orderForm.isInitialized) {
+            orderForm = MutableLiveData()
+            orderForm.value = OrderForm(
+                    TradeItSDKHolder.getSymbolProvider().getJapanSymbol("8703"),
+                    TradeItSDKHolder.getBuyingPower())
+        }
+        return orderForm
+    }
+
+    fun increaseQuantity() {
+        val value = orderForm.value
+        orderForm.value = value?.apply {
+            orderInfo = orderInfo.copy(quantity = orderInfo.quantity + symbol.lotSize)
+        }
+    }
+
+    fun decreaseQuantity() {
+        val value = orderForm.value
+        if (value?.orderInfo?.quantity == value?.symbol?.lotSize) {
+            // do not decrease further when it's lotsize already
+            return
+        } else {
+            orderForm.value = value?.apply {
+                orderInfo = orderInfo.copy(quantity = orderInfo.quantity - symbol.lotSize)
+            }
+        }
+    }
+
+    fun increasePrice() {
+        val value = orderForm.value
+        orderForm.value = value?.apply {
+            if (orderInfo.limitPrice < symbol.priceUpperLimit) {
+                orderInfo = orderInfo.copy(limitPrice = orderInfo.limitPrice + 1)
+            }
+        }
+    }
+
+    fun decreasePrice() {
+        val value = orderForm.value
+        orderForm.value = value?.apply {
+            if (orderInfo.limitPrice > symbol.priceLowerLimit) {
+                orderInfo = orderInfo.copy(limitPrice = orderInfo.limitPrice - 1)
+            }
+        }
+    }
+
+    fun init(symbol: String?) {
+        // TODO need a cleaner way to initialize the OrderForm
```

Comment:
when screen rotates, Fragment will call init again. however we don't need to init the `ViewModel` since we should reuse the model in this case. I'm still looking if there is a way to call this init only once and when it's needed. (need to dig into how Fragment works for this)

Reply:
When configuration changes the fragment is destroyed and then recreated by the fragment manager. However the ViewModel survives configuration changes because you register them into the `ViewModelProviders`, so when the activity is recreated and you call the ViewModelProviders [here](https://github.com/tradingticket/AndroidSDK/pull/108/files/756ec0266165ef8ac58d746aa1404050bee19f66#diff-ec94a0658c37c5aec59e8fb3c425a0baR37) you actually get the same instance. https://developer.android.com/topic/libraries/architecture/viewmodel#lifecycle

Reply:
the view model factory works well. removed the init method now.